R/diagnostics.R

```diff
@@ -6,59 +6,73 @@
 #' `threshold` value, or plot observation indexes vs. diagnostic estimates.
 #' The **Details** section below provides a brief overview of the
 #' diagnostics, but we recommend consulting Vehtari, Gelman, and Gabry (2017)
-#' and Vehtari, Simpson, Gelman, Yao, and Gabry (2019) for full details.
+#' and Vehtari, Simpson, Gelman, Yao, and Gabry (2022) for full details.
 #'
 #' @name pareto-k-diagnostic
 #' @param x An object created by [loo()] or [psis()].
 #' @param threshold For `pareto_k_ids()`, `threshold` is the minimum \eqn{k}
-#'   value to flag (default is `0.5`). For `mcse_loo()`, if any \eqn{k}
-#'   estimates are greater than `threshold` the MCSE estimate is returned as
-#'   `NA` (default is `0.7`). See **Details** for the motivation behind these
-#'   defaults.
+#'   value to flag (default is `1 - 1 / log10(S)`). For `mcse_loo()`, if any
+#'   \eqn{k} estimates are greater than `threshold` the MCSE estimate is
+#'   returned as `NA` (default is `1 - 1 / log10(S)`). See **Details** for
+#'   the motivation behind these defaults.
 #'
 #' @details
 #' The reliability and approximate convergence rate of the PSIS-based estimates
 #' can be assessed using the estimates for the shape parameter \eqn{k} of the
 #' generalized Pareto distribution:
-#' * If \eqn{k < 0.5} then the distribution of raw importance ratios has
-#'   finite variance and the central limit theorem holds. However, as \eqn{k}
-#'   approaches \eqn{0.5} the RMSE of plain importance sampling (IS) increases
-#'   significantly while PSIS has lower RMSE.
 #'
-#' * If \eqn{0.5 \leq k < 1}{0.5 <= k < 1} then the variance of the raw
-#'   importance ratios is infinite, but the mean exists. TIS and PSIS estimates
-#'   have finite variance by accepting some bias. The convergence of the
-#'   estimate is slower with increasing \eqn{k}.
-#'   If \eqn{k} is between 0.5 and approximately 0.7 then we observe practically
-#'   useful convergence rates and Monte Carlo error estimates with PSIS (the
-#'   bias of TIS increases faster than the bias of PSIS). If \eqn{k > 0.7} we
-#'   observe impractical convergence rates and unreliable Monte Carlo error
-#'   estimates.
+#' * If \eqn{k < min(1 - 1 / log10(S), 0.7)}, where \eqn{S} is the
+#'   sample size, PSIS estimate and the corresponding Monte Carlo
+#'   standard error estimate are reliable.
 #'
-#' * If \eqn{k \geq 1}{k >= 1} then neither the variance nor the mean of
-#'   the raw importance ratios exists. The convergence rate is close to
-#'   zero and bias can be large with practical sample sizes.
+#' * If \eqn{1 - 1 / log10(S) <= k < 0.7}, PSIS estimate and the
+#'   corresponding Monte Carlo standard error estimate are not reliable,
+#'   but increasing (effective) sample size \eqn{S} above 2200 may help
+#'   (this will increase the sample size specific threshold
+#'    \eqn{(1-1/log10(2200)>0.7} and then the bias specific threshold
+#'    0.7 dominates).
```

Comment:
```suggestion
#' * If \eqn{1 - 1 / log10(S) <= k < 0.7}, the PSIS estimate and the
#'   corresponding Monte Carlo standard error estimate are not reliable,
#'   but increasing the (effective) sample size \eqn{S} above 2200 may help
#'   (this will increase the sample size specific threshold,
#'    \eqn{1-1/log10(2200)>0.7}, and then the bias specific threshold
#'    0.7 dominates).
```